### PR TITLE
Accept plain objects in js type decls

### PR DESCRIPTION
**Pull request by gw000**
Instead of requiring `Map` objects. For example:
```javascript
oso.registerClass(Foo, {
  types: { id: Number }
});
```
PR checklist:

### CODE REVIEW

File: languages/js/src/Host.ts [hunk 1]
```diff
@@ -159,13 +159,18 @@ export class Host {
...
+    if (!(fields instanceof Map)) fields = new Map(Object.entries(fields));
+
     const userType = new UserType({
       name: clsName,
       cls: cls,
       buildQuery: buildQuery || this.buildQuery,
       execQuery: execQuery || this.execQuery,
       combineQuery: combineQuery || this.combineQuery,
-      fields: types || new Map(),
+      fields: fields,
```
Comment (gj) [inline languages/js/src/Host.ts:173]: I think this works in JS? Been bouncing between all 6 langs this week so my head's a bit spinny

```suggestion
      fields,
```

if so could do it for `cls: cls,` above, too

File: languages/js/src/dataFiltering.test.ts [hunk 2]
```diff
@@ -132,34 +132,34 @@ async function fixtures() {
...
-  numType.set('fooId', String);
-  numType.set('foo', new Relation('one', 'Foo', 'fooId', 'id'));
   oso.registerClass(Num, {
-    types: numType,
     buildQuery: fromRepo(nums, 'num'),
+    types: {
+      number: Number,
+      fooId: String,
+      foo: new Relation('one', 'Foo', 'fooId', 'id'),
+    },
```
Comment (gj) [inline languages/js/src/dataFiltering.test.ts:162]: don't have enough context to approve the PR, but this looks like a big syntactic improvement

**Review by samscott89 — COMMENTED** [review]
Thank you! This was one of my biggest complaints.

I'm a little surprised there's not a change needed to Polar.ts. Doesn't `registerClass` take in a `Map<string, any>`? Is typescript okay with object being passed in as a map?

File: languages/js/src/Host.ts (hunk 1 above)
Comment (gw000) [reply languages/js/src/Host.ts:173]: Oooh let me check, I sure hope that works...

Comment (gw000) [reply languages/js/src/Host.ts]: 👍 thanks!

**Comment by gw000** [comment]
@samscott89 `registerClass` used to work that way but the number of parameters got too large to manage positionally, so now it takes an optional object of params to simulate keyword arguments. `UserType` still requires a map but `cacheClass` makes sure it always gets one.

File: languages/js/src/Polar.ts [hunk 3]
```diff
@@ -254,6 +254,14 @@ export class Polar {
 
   /**
    * Register a JavaScript class for use in Polar policies.
+   *
+   * @param cls The class to register.
+   * @param params An optional object with extra parameters.
+   *
+   * Accepted extra parameters are:
```
Comment (samscott89) [inline languages/js/src/Polar.ts:261]: Can we make a follow up issue to go back and add types for these?

Comment (samscott89) [reply languages/js/src/Polar.ts:261]: Or an interface, whatever the right typescript-ism is

Comment (gw000) [reply languages/js/src/Polar.ts:261]: Added it!

**Review by samscott89 — APPROVED** [review]
This looks good! Thanks @gw000! and thanks @gj for the TS suggestions :)

**Pull request merged**